### DISCUSSION TITLE
How to specify OpenAPI input document for C# client generation in nswag.json

### DISCUSSION THREAD

**Discussion by BBreiden** [RicoSuter/NSwag#4089] [Q&A]
I would like to generate a C# client from a given OpenAPI document using the "run" command on the command line. I have tried the following:

- Specifying /input on the command line (like for the openapitocsharp command) seems not to work
- Adding an "input" value in the openApiToCSharpClient section of the nswag file also did not do the trick

Looking forward to suggestions! :-)

**Comment by reiv** [comment]
Raising this one from the dead in case anyone else stumbles upon the same issue:

You want a `documentGenerator` section with `fromDocument` in your `nswag.json`, like this:

```json5
{
  "runtime": "Net90",
  "documentGenerator": {
    "fromDocument": {
      "url": "swagger.json"
    }
  },
  "codeGenerators": {
    // e.g. openApiToCSharpClient
  }
}

```

I could not find this documented *anywhere* (and I grepped through the wiki to double-check - no mention of `fromDocument`).

**Reply by lawsonjl-ornl** [reply]
@reiv Thank you for posting this. I just spent several hours trying to find this information _anywhere_ in the repository, to the point of following definitions/references around in the github source explorer trying to track down where some input file loader originally got its config from - with no luck.

It's crazy this isn't documented at all. How does anyone even manage to use this feature?
